atlas_provider_django/management/commands/atlas-provider-django.py

```diff
@@ -101,17 +52,84 @@ def get_connection_by_dialect(dialect):
             }, "sqlite3")
             conn.SchemaEditorClass = MockSqliteSchemaEditor
         case Dialect.postgresql:
+            from django.db.backends.postgresql.base import DatabaseWrapper as PGDatabaseWrapper
+            from django.db.backends.postgresql.schema import DatabaseSchemaEditor as PGDatabaseSchemaEditor
+
+            class MockPGDatabaseSchemaEditor(PGDatabaseSchemaEditor):
+                def __init__(self, *args, **kwargs):
+                    super().__init__(*args, **kwargs)
+
+                def execute(self, sql, params=()):
+                    return super(PGDatabaseSchemaEditor, self).execute(sql, params)
+
             conn = PGDatabaseWrapper({
                 "ENGINE": "django.db.backends.postgresql",
             }, "postgresql")
             conn.SchemaEditorClass = MockPGDatabaseSchemaEditor
         case Dialect.mysql:
+            from django.db.backends.mysql.base import DatabaseWrapper as MySQLDatabaseWrapper
+            from django.db.backends.mysql.schema import DatabaseSchemaEditor as MySQLDatabaseSchemaEditor
+            from django.db.backends.mysql.features import DatabaseFeatures as MySQLDatabaseFeatures
+
+            class MockMySQLDatabaseSchemaEditor(MySQLDatabaseSchemaEditor):
+                def __init__(self, *args, **kwargs):
+                    super().__init__(*args, **kwargs)
+
+                # Override the method of MySQLDatabaseSchemaEditor since it checks the storage engine.
+                # We assume that the storage engine is InnoDB.
+                def _field_should_be_indexed(self, model, field):
+                    if not super(MySQLDatabaseSchemaEditor, self)._field_should_be_indexed(model, field):
+                        return False
+                    if field.get_internal_type() == "ForeignKey" and field.db_constraint:
+                        return False
+                    return not self._is_limited_data_type(field)
+
+                def _supports_limited_data_type_defaults(self):
+                    return True
+
+            class MockMySQLDatabaseFeatures(MySQLDatabaseFeatures):
+                def __init__(self, *args, **kwargs):
+                    super().__init__(*args, **kwargs)
+
+                def has_native_uuid_field(self):
```

Comment:
This is besically checking if the mysql driver is 8 or lower, right?

Reply:
This is true, but In order to determine the mysql version I need to call the DB (what I cannot do). so I mock this method

Reply:
We should think about maybe adding the target version to the query string / arguments in addition to the dialect

Reply:
Thats good point, but lets discuss is in seperate PR